test/integration/src/features/margin.py

```diff
@@ -160,5 +169,56 @@ def test_swap_external_to_external(self):
             assert True  # no-op line just for setting a breakpoint
 
     def test_swap(self):
-        self.test_swap_rowan_to_external()
+        # self.test_swap_rowan_to_external()
         # self.test_swap_external_to_external()
+        self.test_margin()
+
+    def test_margin(self):
+        borrow_asset = "rowan"
+        collateral_asset = "cusdc"
+        collateral_amount = 10**20
+        leverage = 2
+
+        with self.with_test_env(self.default_pool_setup) as (env, sifnoded, pools_definitions):
+            account = sifnoded.create_addr()
+            env.fund(account, {
+                ROWAN: 10**25,
+                collateral_asset: 10**25,
+            })
+            margin_params = sifnoded.query_margin_params()
+
+            # sifnoded.tx_margin_whitelist(env.clp_admin, account, broadcast_mode="block")
+
+            pool_before_open = sifnoded.query_pools_sorted()[collateral_asset]
```

Comment:
variable is never used. I think we can make assertions against `pool_after_open` ?